win32ss/drivers/videoprt/videoprt.c

```diff
@@ -419,34 +419,34 @@ IntVideoPortFindAdapter(
     PVIDEO_PORT_DEVICE_EXTENSION DeviceExtension;
     NTSTATUS Status;
     VP_STATUS vpStatus;
-    VIDEO_PORT_CONFIG_INFO ConfigInfo;
+    PVIDEO_PORT_CONFIG_INFO ConfigInfo;
```

Comment:
Any reason to change the variable to a pointer here? You storing the memory address of the `VIDEO_PORT_CONFIG_INFO` variable from `VIDEO_PORT_DEVICE_EXTENSION` at line 429, not even a pointer, and you don't do any allocation from dynamical memory. You can simply pass its memory address via the `&` operator where it is needed, as it was done before. So it seems needless.

Reply:
I don't get your review comment. The memory address of the variable is a pointer by definition. He changed the structure on stack to the pointer and then using it.

Reply:
> He changed the structure on stack to the pointer and then using it.

@binarymaster: but he didn't do it in the device extension. Just look at line 113 in videoprt.h:
![image](https://github.com/reactos/reactos/assets/26385117/83483b96-7cef-428d-96ef-f93a295deeaf)
It's declared not as `PVIDEO_PORT_CONFIG_INFO`, simply as `VIDEO_PORT_CONFIG_INFO` instead. So it's not a pointer and cannot be such, otherwise it also will need the additional allocation/freeing.
Then, he simply assigns its memory address via `&` to the locally declared pointer:
![image](https://github.com/reactos/reactos/assets/26385117/9ad00cda-25c3-40d5-9855-9292b388a3cd)
It makes sense only when assigning pointer to pointer, e. g.
```
PVIDEO_PORT_CONFIG_INFO ConfigInfo;
ConfigInfo = DeviceExtension->ConfigInfo;
```
in case `DeviceExtension->ConfigInfo` is a pointer too. But it actually isn't.
My point is, it's not necessary needed, and with my suggestion, I simply mean the following:
```
// declare it as a normal variable instead:
VIDEO_PORT_CONFIG_INFO ConfigInfo;
// then assign it without & operator:
ConfigInfo = DeviceExtension->ConfigInfo;
// use it as normal variable (with & and use . instead of ->):
    RtlZeroMemory(&ConfigInfo, sizeof(VIDEO_PORT_CONFIG_INFO));
    ConfigInfo.Length = sizeof(VIDEO_PORT_CONFIG_INFO);
    ConfigInfo.AdapterInterfaceType = DeviceExtension->AdapterInterfaceType;
    if (ConfigInfo.AdapterInterfaceType == PCIBus)
        ConfigInfo.InterruptMode = LevelSensitive;
// and pass its memory address via the & in the calls which require the pointer:
            vpStatus = DriverExtension->InitializationData.HwFindAdapter(
                         &DeviceExtension->MiniPortDeviceExtension,
                         DriverExtension->HwContext,
                         NULL,
                         &ConfigInfo,
                         &Again);
if (!IntVideoPortSetupInterrupt(DeviceObject, DriverExtension, &ConfigInfo))
```
Like it was done before.
I guess it's needless using a pointer when doing that is not required at all.
Everything other in this PR is fine to me.

Reply:
> It's declared not as PVIDEO_PORT_CONFIG_INFO, simply as VIDEO_PORT_CONFIG_INFO instead. So it's not a pointer

That's right, it's not a pointer, but the structure data, and it is stored within the extension data (on the stack or whatever).

> So it's not a pointer and cannot be such, otherwise it also will need the additional allocation/freeing.

You access the data by memory address, which IS a pointer and which could be assigned to a pointer variable of the same type.

> Then, he simply assigns its memory address via & to the locally declared pointer

Yes, and it totally makes sense.

> It makes sense only when assigning pointer to pointer

But again, the memory address is what a pointer contains. Please update your knowledge base.

> ```c
> // declare it as a normal variable instead:
> VIDEO_PORT_CONFIG_INFO ConfigInfo;
> // then assign it without & operator:
> ConfigInfo = DeviceExtension->ConfigInfo;
> ```

This is not permitted in C language, you cannot assign a struct to struct. You either assign struct members manually or use memcpy / RtlCopyMemory.

Reply:
It's an alias. A "shortcut" name. So you can write `ConfigInfo` instead of `&DeviceExtension->ConfigInfo`.
That is all there is to it.

Reply:
> This is not permitted in C language, you cannot assign a struct to struct. You either assign struct members manually or use memcpy / RtlCopyMemory.

That is simply wrong.

Reply:
Alright, but then I don't see a reason to keep structs in both places. Using a pointer (or a "shortcut" as you called it) in the function makes perfect sense and avoids duplicating the struct.

Reply:
That's why I used a pointer. Anyways, i tried to keep my code the most tiny as possible